bin/resin-preload

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env node
+const info = require('../package.json')
+
+const USAGE = `
+  Usage: ${info.name} [options]
+
+  Options:
+
+    --app          Application ID (required)
+    --img          Disk image to preload into (required)
+    --api-token    API token (required, or api-key)
+    --api-key      API key (required, or api-token)
+    --api-host     API host (default: "https://api.resin.io")
+    --registry     Image registry host (default: "registry.resin.io")
+
+    --help, -h     Display ${info.name} usage
+    --version, -v  Display ${info.name} version
+
+  Environment variables:
+
+    The following option flags can also be set
+    via the corresponding environment variables:
+
+    --app          APP_ID
+    --img          IMAGE
+    --api-token    API_TOKEN
+    --api-key      API_KEY
+    --api-host     API_HOST
+    --registry     REGISTRY_HOST
+
+  Example:
+
+    ${info.name} --app 123456 --api-token \"xxxx...\" --img /path/to/resin-os.img
+`
+
+function inspect( value ) {
+  return require('util').inspect( value, {
+    colors: process.stdout.isTTY
+  })
+}
+
+function showError( error, details, report ) {
+
+  let message = !details ?
+    `\n[ERROR] ${error.message}\n` :
+    `\n${inspect(error)}\n`
+
+  if(report) {
+    message += `\nLooks like this might be an issue with ${info.name};\n`
+    message += `Please report it at ${info.bugs.url}\n`
+  }
+
+  process.stderr.write(message)
+
+}
+
+process.on('uncaughtException', (error) => {
+  const code = typeof error.code === 'number' ?
+    (error.code || 1) : 1
+  showError(error, true, true)
+  process.exit(code)
+})
+
+let argv = process.argv.slice(2)
+
+if( argv.includes('--help') || argv.includes('-h') ) {
+  process.stderr.write(USAGE)
+  process.exit(0)
+}
+
+if( argv.includes('--version') || argv.includes('-v') ) {
+  process.stdout.write(info.version + '\n')
+  process.exit(0)
+}
+
+const options = {
+  appId: process.env['APP_ID'],
+  image: process.env['IMAGE'],
+  apiToken: process.env['API_TOKEN'],
+  apiKey: process.env['API_KEY'],
+  apiHost: process.env['API_HOST'],
+  registryHost: process.env['REGISTRY_HOST'],
+}
+
+while( argv.length ) {
```

Comment:
Have you looked into an arg parsing module like [capitano](https://www.npmjs.com/package/capitano) or [commander](https://www.npmjs.com/package/commander)